landlab/components/weathering/exponential_weathering.py

```diff
@@ -133,8 +133,12 @@ def __init__(
     def calc_soil_prod_rate(self):
         """Calculate soil production rate."""
         # apply exponential function
-        self._soil_prod_rate[self._grid.core_nodes] = self._w0 * np.exp(
-            -self._depth[self._grid.core_nodes] / self._wstar
+        w0_vector = np.broadcast_to(self._w0, self._depth.shape)
+        wstar_vector = np.broadcast_to(self._wstar, self._depth.shape)
+
+        core = self._grid.core_nodes
+        self._soil_prod_rate[core] = w0_vector[core] * np.exp(
+            -self._depth[core] / wstar_vector[core]
```

Comment:
Since this is the only place you use the *soil_prod_rate* and *depth* fields, I would just refer to the fields directly.

i.e.
```python
self.grid.at_node["soil_production__rate"][core] = self._w0[core] * np.exp(
    -self.grid.at_node["soil__depth"][core] / self._wstar
)
```
Apart from having fewer class variables, this also has the advantage that it ensures you are using the current values for these fields even if another component accidentally changes what those fields point to.

Reply:
Yeah, agreed. This was inherited from the previous version. Adjusted now.